GeeksCoreLibrary/Core/Extensions/DataRowExtensions.cs

```diff
@@ -129,16 +133,23 @@ public static JObject ToJsonObject(this DataRow dataRow, string requiredColumnNa
 
                     result.TryAdd(TrimPrefix(columnName[..^(withDate ? EncryptListWithDateSuffix : EncryptListSuffix).Length], requiredColumnNamePrefix), value);
                 }
-                else if (columnName.EndsWith(NormalEncryptSuffix, StringComparison.OrdinalIgnoreCase) || columnName.EndsWith(NormalEncryptWithDateSuffix, StringComparison.OrdinalIgnoreCase))
+                else if (columnName.EndsWith(NormalEncryptSuffix, StringComparison.OrdinalIgnoreCase) || columnName.EndsWith(NormalEncryptWithDateSuffix, StringComparison.OrdinalIgnoreCase) || columnName.EndsWith(SaferEncryptSuffix, StringComparison.OrdinalIgnoreCase) || columnName.EndsWith(SaferEncryptWithDateSuffix, StringComparison.OrdinalIgnoreCase))
                 {
-                    var withDate = columnName.EndsWith(NormalEncryptWithDateSuffix, StringComparison.OrdinalIgnoreCase);
+                    var withDate = columnName.EndsWith(NormalEncryptWithDateSuffix, StringComparison.OrdinalIgnoreCase) || columnName.EndsWith(SaferEncryptWithDateSuffix, StringComparison.OrdinalIgnoreCase);
+                    var saferMethod = columnName.EndsWith(SaferEncryptSuffix, StringComparison.OrdinalIgnoreCase) || columnName.EndsWith(SaferEncryptWithDateSuffix, StringComparison.OrdinalIgnoreCase);
                     var value = Convert.ToString(dataRow[dataColumn]);
                     if (!String.IsNullOrWhiteSpace(value))
                     {
-                        value = value.EncryptWithAes(encryptionKey, withDate);
+                        value = value.EncryptWithAes(encryptionKey, withDate, useSlowerButMoreSecureMethod:saferMethod);
+                    }
+                    if (saferMethod)
+                    {
+                        result.TryAdd(TrimPrefix(columnName[..^(withDate ? SaferEncryptWithDateSuffix : SaferEncryptSuffix).Length], requiredColumnNamePrefix), value.UrlEncode());
+                    }
+                    else
+                    {
+                        result.TryAdd(TrimPrefix(columnName[..^(withDate ? NormalEncryptWithDateSuffix : NormalEncryptSuffix).Length], requiredColumnNamePrefix), value.UrlEncode());
```

Comment:
What is all of this? This seems to have nothing to do with loading assembly for NoPSP.